src/Features/CSharp/Portable/DocumentSymbols/CSharpDocumentSymbolsService.cs

```diff
@@ -0,0 +1,248 @@
+                         IMethodSymbol { MethodKind: MethodKind.PropertyGet or MethodKind.PropertySet }:
+                        continue;
+
+                    case IMethodSymbol { PartialImplementationPart: { } partialPart }:
+                        membersBuilder.Add(CreateInfo(member, tree, declarationService, ImmutableArray<DocumentSymbolInfo>.Empty, cancellationToken));
+                        membersBuilder.Add(CreateInfo(partialPart, tree, declarationService, ImmutableArray<DocumentSymbolInfo>.Empty, cancellationToken));
+                        break;
+
+                    default:
+                        membersBuilder.Add(CreateInfo(member, tree, declarationService, ImmutableArray<DocumentSymbolInfo>.Empty, cancellationToken));
+                        break;
+                }
+            }
+
+            membersBuilder.Sort((d1, d2) => d1.Text.CompareTo(d2.Text));
+            return CreateInfo(type, tree, declarationService, membersBuilder.ToImmutable(), cancellationToken);
+        }
+
+        protected override DocumentSymbolInfo CreateInfo(
+            ISymbol symbol,
+            SyntaxTree tree,
+            ISymbolDeclarationService declarationService,
+            ImmutableArray<DocumentSymbolInfo> childrenSymbols,
+            CancellationToken cancellationToken)
+        {
+            var enclosingSpans = GetEnclosingSpansForSymbol(symbol, tree, declarationService);
+            var declaringSpans = GetDeclaringSpans(symbol, tree);
+            var obsolete = symbol.GetAttributes().Any(attr => attr.AttributeClass?.MetadataName == "ObsoleteAttribute");
+
+            var isConstant = symbol is IFieldSymbol { IsConst: true } or ILocalSymbol { IsConst: true };
+
+            return new(
+                symbol.ToDisplayString(symbol is ITypeSymbol ? s_typeFormat : s_memberFormat),
+                symbol.Name,
+                symbol.GetGlyph(),
+                obsolete,
+                isConstant ? s_constantTag : ImmutableArray<string>.Empty,
+                DocumentSymbolInfoExtensions.GetDocumentSymbolInfoPropertiesForSymbol(symbol, cancellationToken),
+                enclosingSpans,
+                declaringSpans,
+                childrenSymbols);
+        }
+
+        protected override bool ConsiderNestedNodesChildren(ISymbol node) => node is not (ILabelSymbol or ILocalSymbol);
+
+        private static ImmutableArray<TextSpan> GetEnclosingSpansForSymbol(ISymbol symbol, SyntaxTree tree, ISymbolDeclarationService declarationService)
+        {
+            using var _ = ArrayBuilder<TextSpan>.GetInstance(out var spans);
+
+            if (symbol.Kind is SymbolKind.Field or SymbolKind.Local)
+            {
+                if (symbol.ContainingType.TypeKind == TypeKind.Enum)
+                {
+                    AddEnumMemberSpan(symbol, tree, spans);
+                }
+                else
+                {
+                    AddFieldOrLocalSpan(symbol, tree, spans);
+                }
```

Comment:
all these different cases are hard to read.  ie. if it's a field or a local, except in teh case where it's an enum, otherwise do this.

Can you just break out hte separate cases and deal with each one individually?